src/main/java/com/example/security/JWTAuthenticationFilter.java

```diff
@@ -12,6 +12,13 @@
 
 public class JWTAuthenticationFilter extends GenericFilterBean {
 
+    private TokenAuthenticationService tokenAuthenticationService;
+
+    @Autowired
```

Comment:
`@Autowired` on constructors are used to inject instances through the constructor's parameters. In this case, you are not injecting anything, because there is no parameters expected by this constructor.

Furthermore, the `TokenAuthenticationService` has only static methods and therefore doesn't need to be instantiated.